libcontainer/cgroups/systemd/common.go

```diff
@@ -160,7 +160,15 @@ func findDeviceGroup(ruleType devices.Type, ruleMajor int64) (string, error) {
 
 // generateDeviceProperties takes the configured device rules and generates a
 // corresponding set of systemd properties to configure the devices correctly.
-func generateDeviceProperties(rules []*devices.Rule) ([]systemdDbus.Property, error) {
+func generateDeviceProperties(r *configs.Resources) ([]systemdDbus.Property, error) {
+	if r.SkipDevices {
+		// Not adding any Device* properties means the default of
+		// DevicePolicy=auto will be used by systemd, which allows
+		// access to all devices (if no explicit DeviceAllow= is
+		// present).
+		return []systemdDbus.Property{}, nil
```

Comment:
Does this actually reset the policy or does it keep the old one? What I did was set `DevicePolicy=auto` explicitly later in this function. If that's really not necessary then we can also drop it from the later "this rule is allow-all" `return ...`.

Reply:
Initially I did the same thing but then thought it is not needed. Now I'm not sure -- will test this tomorrow.

Reply:
Ah, OK, I did not have the update scenario in mind, only the initial one-time configuration (assuming no `Device*` properties were ever set before for the given unit).

Modified the commit to explicitly return `DevicePolicy=auto` and an empty list of `DeviceAllow` rules, reusing the code you added earlier.